python/ola/PidStore.py

```diff
@@ -851,7 +895,7 @@ def GetDescription(self, indent=0):
       names.append('<%s>' % atom.name)
       output.append(atom.GetDescription(indent=2))
 
-    return ' '.join(names), '\n'.join(output)
+    return ' '.join(names) + ':\n' + '\n'.join(output)
```

Comment:
I want to call out this change b/c it's an interface change that changes the return type of GetDescription from a tuple to a string.  I couldn't find anything else in the code base that would depend on that and doubt it was deliberate, but I could be wrong.  However, in addition to making more sense to return a string, the tuple implementation broke for Groups inside a Group (b/c it wound up trying to convert a tuple to a string) so needed to have a string here.

Reply:
I think this is equivalent to this in C++ land? https://github.com/OpenLightingProject/ola/blob/master/common/rdm/PidStoreHelper.cpp#L227-L235

A group within a group would certainly be at the edge of expected behaviour based on this:
https://wiki.openlighting.org/index.php/RDM_PID_Definitions#Parameter_Definitions_and_OLA

The inner group would need to have a defined length to allow the outer group length to be calculated. I guess if you had say a set of curves, with each curve having five points, it might make sense.

From a technical perspective, why have you added a : before the newline? Can I see some example output from this?

Actually looking further, I think this will break a command, but due to some admittedly rather lacking documentation. If you run the following command:
`./python/examples/ola_rdm_get.py -u 0 --uid 7a70:ffffff00 sensor_definition`

What output do you get? I get the following, I don't think yours will match anymore:
```
Usage: get sensor_definition <sensor_number>
  sensor_number: <[0, 254]> 

*** Missing argument for sensor_number
```

Reply:
Perhaps we should add a test for this too?

Reply:
Ah, I didn't know the use case for this.  In some ideal world I think I would expect a usage message to ask separately for the summary of parameters and then details, and also for the python to display e.g. uint32 like the C++ does instead of the numbers, but I'd rather not dive into that sort of refactor at this point :)

Reply:
I only found it after a bit of hunting through the code.

Do you mean have two separate methods which essentially both walk the tree? So in simple terms add two methods which both call GetDescription and return the two arguments individually?

Yeah definitely another one for a separate PR! The Python is returning the valid range (so in the case of sensor_number it's not uint8, as you can't get 0xFF for that particular PID; as it means ALL_SENSORS, and getting all doesn't make sense). But we could look at those other changes and making them more similar.